apps/dotcom/client/src/tla/providers/TlaAppProvider.tsx

```diff
@@ -1,11 +1,68 @@
+import { getAssetUrlsByImport } from '@tldraw/assets/imports.vite'
+import { useCallback, useState } from 'react'
 import { Outlet } from 'react-router-dom'
-import { AppStateProvider } from '../hooks/useAppState'
+import {
+	AssetUrlsProvider,
+	ContainerProvider,
+	TLUiEventHandler,
+	TldrawUiDialogs,
+	TldrawUiDialogsProvider,
+	TldrawUiEventsProvider,
+	TldrawUiToasts,
+	TldrawUiToastsProvider,
+	TldrawUiTranslationProvider,
+	useValue,
+} from 'tldraw'
+import { AppStateProvider, useApp } from '../hooks/useAppState'
 import '../styles/tla.css'
 
+export const assetUrls = getAssetUrlsByImport()
+
 export function Component() {
 	return (
 		<AppStateProvider>
-			<Outlet />
+			<Inner />
 		</AppStateProvider>
 	)
 }
+
+function Inner() {
+	const app = useApp()
+	const theme = useValue('theme', () => app.getSessionState().theme, [app])
+	const [container, setContainer] = useState<HTMLElement | null>(null)
+
+	return (
+		<div
+			ref={setContainer}
+			className={`tla tl-container ${theme === 'light' ? 'tla-theme__light tl-theme__light' : 'tla-theme__dark tl-theme__dark'}`}
+		>
+			{container && (
+				<ContainerProvider container={container}>
+					<InnerInner />
+				</ContainerProvider>
+			)}
+		</div>
+	)
+}
+
+function InnerInner() {
+	const handleAppLevelUiEvent = useCallback<TLUiEventHandler>(() => {
+		// todo, implement handling ui events at the application layer
+	}, [])
+
+	return (
+		<AssetUrlsProvider assetUrls={assetUrls}>
+			<TldrawUiEventsProvider onEvent={handleAppLevelUiEvent}>
+				<TldrawUiTranslationProvider locale="en">
```

Comment:
Do we need this here? We'll probably have a different setup for app translations. We can then also keep the editor inside the provider. We also wouldn't needs `AssetUrlsProvider` then.  Unless we are reusing some of the translations?

Reply:
That's true. I added translations here from a "might as well" philosophy. The assets are important however for icons that are used in the UI components.

Reply:
I don't think we need assets. Tldraw will add the provider anyway, so we are kinda nesting providers.

Reply:
Without assets, the TldrawUiIcon component fails. Its ok to nest react contexts (we'll already be doing it for toasts, dialogs, etc, too)

Reply:
`TldrawUiIcon` is not used from tla at the moment. And for the usages in the tldraw package we already use the assets provider there.